### DISCUSSION TITLE
High memory usage on fly.io deployments

### DISCUSSION THREAD

**Discussion by doloresjose** [benbusby/whoogle-search#527] [Q&A]
The docker image gets deployed successfully and the landing page is loaded, but when a search request is sent it just crashes and throws ERROR 502.
![image](https://user-images.githubusercontent.com/62073313/140419491-97b6de69-1562-4ee1-9ffa-23b2a0d5fb9d.png)

I think due to the limited ram availability the app crashes i am using v0.6.0, a screen cap of ram usage and app log is attached below.
![image](https://user-images.githubusercontent.com/62073313/140419715-e395f523-a1a6-44b6-96b8-1dd62a60b63a.png)

[APP LOGS](https://pastebin.mozilla.org/ANtg9Mfb)

**Comment by benbusby** [comment]
That seems like a weird amount of memory used by a docker deployment. Running the container locally the app peaks at ~70MB, so I'm not sure why an instance deployed with fly.io is using this much memory. It should definitely run comfortably within the 256MB limit in fly.io's free tier. 

Are you running more than one app on fly.io? That could potentially cause issues since memory is shared between multiple apps. Otherwise I'm at a loss for why it would be using so much more memory on their platform than elsewhere on other platforms. 

Moving to a discussion for now, since this seems to potentially be an issue on fly.io's side.

**Comment by kevinfiol** [comment]
@doloresjose Would you be able to share your `fly.toml` (with sensitive details removed)? I haven't had luck deploying 0.7.1 to fly.io with the README's instructions.
